### PR TITLE
Sage 300: Complete Export Log [PR Comments] - 2.1

### PR DESCRIPTION
**Pull request by anishfyle**
PR Comments and Fixes for #299 

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `FundSource` enum to categorize funding sources.
  - Implemented a new search functionality in the app's interface.

- **Enhancements**
  - Improved the accounting export interface to include additional filtering options such as date ranges and pagination.
  - Streamlined the process of importing expenses from Fyle with a new dedicated service.

- **Bug Fixes**
  - Corrected the display of enum values to ensure they are properly formatted for end-user readability.

- **Refactor**
  - Consolidated accounting export-related operations into a new `AccountingExportService`.
  - Removed redundant code and methods from various services to centralize functionality.

- **Style**
  - Updated the user interface for the export log filter and table components for a more intuitive user experience.

- **Documentation**
  - Updated component documentation to reflect changes in the import statements and method usage.

- **Chores**
  - Removed unnecessary newline characters from model files to clean up the codebase.

- **Revert**
  - Reverted event binding names to align with updated component methods.

### CODE REVIEW

File: src/app/core/services/common/export-log.service.ts [hunk 1]
```diff
@@ -56,4 +105,26 @@ export class ExportLogService {
     return `${url}?org_id=${this.org_id}`;
   }
 
+
+  // GetAccountingExports(state: AccountingExportStatus | AccountingExportStatus.COMPLETE, limit: number, offset: number, selectedDateFilter: SelectedDateFilter | null, exportedAt?: Date | null): Observable<AccountingExportResponse> {
```
Comment (ashwin1111) [inline src/app/core/services/common/export-log.service.ts:109]: remove commented code

File: src/app/core/services/common/export-log.service.ts [hunk 2]
```diff
@@ -23,6 +26,52 @@ export class ExportLogService {
...
+
+    return (
+      employeeName.toLowerCase().includes(query) ||
+      employeeID.toLowerCase().includes(query) ||
+      expenseType.toLowerCase().includes(query) ||
+      referenceNumber.toLowerCase().includes(query)
+    );
+  }
+
   getReferenceType(description: Partial<ExpenseGroupDescription>): FyleReferenceType {
```
Comment (ashwin1111) [inline src/app/core/services/common/export-log.service.ts:75]: Partial is not required

File: src/app/integrations/sage300/sage300-main/sage300-export-log/sage300-complete-export-log/sage300-complete-export-log.component.html [hunk 3]
```diff
@@ -26,28 +26,15 @@
...
-                <img src="assets/illustrations/search-result-empty.svg" width="200px" hight="200px" class="tw-inline-block">
-                <div class="tw-pt-12-px">
-                    <h4 class="tw-font-500 tw-text-14-px tw-text-slightly-normal-text-color">Sorry, no results found!</h4>
-                    <p class="tw-font-400 tw-text-14-px tw-text-faded-text-color tw-pt-8-px">We could not find what you were looking for.<br>Kindly check the keywords again.</p>
-                </div>
-            </div>
-        </div>
+        <app-zero-state-with-illustration *ngIf="!filteredAccountingExports.length"
+            [mainText]="'Sorry, no results found!'"
+            [subText]="'We could not find what you were looking for.Kindly check the keywords again.'">
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-main/sage300-export-log/sage300-complete-export-log/sage300-complete-export-log.component.html:31]: 1 space after .

**Comment by anishfyle** [comment]
@coderabbitai review

File: src/app/integrations/sage300/sage300-main/sage300-export-log/sage300-complete-export-log/sage300-complete-export-log.component.ts [hunk 4]
```diff
@@ -124,69 +92,26 @@ export class Sage300CompleteExportLogComponent implements OnInit {
...
-          fyleReferenceType: '',
-          referenceNumber: referenceNumber,
-          exportedAs: accountingExport.type,
-          fyleUrl: this.exportLogService.generateFyleUrl(accountingExport, referenceType),
-          integrationUrl: accountingExport.export_url,
-          expenses: accountingExport.expenses
-        });
+        const accountingExports: AccountingExportList[] = accountingExportResponse.results.map(accountingExport =>
+          AccountingExportUtils.createAccountingExport(accountingExport, this.exportLogService)
+        );
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/integrations/sage300/sage300-main/sage300-export-log/sage300-complete-export-log/sage300-complete-export-log.component.ts]: just move the for loop, nothing else. This part
const accountingExports: AccountingExportList[] = accountingExportResponse.results.map(accountingExport =>
  AccountingExportUtils.createAccountingExport(accountingExport, this.exportLogService)
);

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**